### PR TITLE
Plugin Example Tutorial Step7

### PR DESCRIPTION
**Pull request by Baptistebgh**
#Plugin Example Tutoriel Step7

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+            <type>lutece-core</type>
+        </dependency>
+        
+        <dependency>
+            <groupId>fr.paris.lutece</groupId>
+            <artifactId>lutece-core</artifactId>
+            <version>6.0.0-SNAPSHOT</version>
+            <type>lutece-core</type>
+        </dependency>
+
```
Comment (jonenst) [inline pom.xml]: lutece-core dependency duplicate

File: .classpath [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+			<attribute name="maven.pomderived" value="true"/>
+		</attributes>
+	</classpathentry>
+	<classpathentry kind="con" path="org.eclipse.m2e.MAVEN2_CLASSPATH_CONTAINER">
+		<attributes>
+			<attribute name="maven.pomderived" value="true"/>
+		</attributes>
+	</classpathentry>
+	<classpathentry kind="output" path="target/classes"/>
+</classpath>
```
Comment (jonenst) [inline .classpath]: we usually don't commit this file

File: .project [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+			<name>org.eclipse.m2e.core.maven2Builder</name>
+			<arguments>
+			</arguments>
+		</buildCommand>
+	</buildSpec>
+	<natures>
+		<nature>org.eclipse.jdt.core.javanature</nature>
+		<nature>org.eclipse.m2e.core.maven2Nature</nature>
+	</natures>
+</projectDescription>
```
Comment (jonenst) [inline .project]: we usually don't commit this file

File: .settings/org.eclipse.core.resources.prefs [hunk 4]
```diff
@@ -0,0 +1,4 @@
+eclipse.preferences.version=1
+encoding//src/java=UTF-8
+encoding//src/test/java=UTF-8
+encoding/<project>=UTF-8
```
Comment (jonenst) [inline .settings/org.eclipse.core.resources.prefs]: we usually don't commit this file

File: .settings/org.eclipse.jdt.core.prefs [hunk 5]
```diff
@@ -0,0 +1,5 @@
+eclipse.preferences.version=1
+org.eclipse.jdt.core.compiler.codegen.targetPlatform=1.8
+org.eclipse.jdt.core.compiler.compliance=1.8
+org.eclipse.jdt.core.compiler.problem.forbiddenReference=warning
+org.eclipse.jdt.core.compiler.source=1.8
```
Comment (jonenst) [inline .settings/org.eclipse.jdt.core.prefs]: we usually don't commit this file

File: .settings/org.eclipse.m2e.core.prefs [hunk 6]
```diff
@@ -0,0 +1,4 @@
+activeProfiles=
+eclipse.preferences.version=1
+resolveWorkspaceProjects=true
+version=1
```
Comment (jonenst) [inline .settings/org.eclipse.m2e.core.prefs]: we usually don't commit this file

File: src/java/fr/paris/lutece/plugins/example/modules/elastic/elasticdata/ElasticProjectDao.java [hunk 7]
```diff
@@ -0,0 +1,39 @@
+package fr.paris.lutece.plugins.example.modules.elastic.elasticdata;
+
+import java.util.ArrayList;
+import java.util.Collection;
+
+import fr.paris.lutece.plugins.elasticdata.business.DataObject;
+import fr.paris.lutece.util.sql.DAOUtil;
+
+public class ElasticProjectDao {
+	 private static final String SQL_QUERY_SELECTALL = "SELECT * FROM lutece_test2.example_project;";
```
Comment (jonenst) [inline src/java/fr/paris/lutece/plugins/example/modules/elastic/elasticdata/ElasticProjectDao.java]: there should not be a reference to the database name

File: src/java/fr/paris/lutece/plugins/example/modules/elastic/business/Project.java [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+
+    /**
+     * Sets the Cout
+     * @param nCout The Cout
+     */ 
+    public void setCout( int nCout )
+    {
+        _nCout = nCout;
+    }
+}
```
Comment (jonenst) [inline src/java/fr/paris/lutece/plugins/example/modules/elastic/business/Project.java]: Why is this business class needed in this project ? How is it different from the Project class in the plugin-example project

File: src/java/fr/paris/lutece/plugins/example/modules/elastic/elasticdata/ProjectDataObject.java [hunk 9]
```diff
@@ -0,0 +1,44 @@
+package fr.paris.lutece.plugins.example.modules.elastic.elasticdata;
+
+import fr.paris.lutece.plugins.elasticdata.business.DataObject;
+
+/**
+ * My DataObject
```
Comment (jonenst) [inline src/java/fr/paris/lutece/plugins/example/modules/elastic/elasticdata/ProjectDataObject.java]: The javadoc should be updated

File: src/java/fr/paris/lutece/plugins/example/modules/elastic/elasticdata/ProjectDataSource.java [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+ * My DataSource
+ */
+public class ProjectDataSource extends AbstractDataSource implements DataSource
+{
+
+    @Override
+    public Collection<DataObject> getDataObjects( )
+    {
+        ElasticProjectDao dao = new ElasticProjectDao();
+        return dao.selectAll();
```
Comment (jonenst) [inline src/java/fr/paris/lutece/plugins/example/modules/elastic/elasticdata/ProjectDataSource.java]: The DAO should be a singleton injected by spring (I see the wiki needs to be updated as well)

File: src/java/fr/paris/lutece/plugins/example/modules/elastic/extend/ExtendProjectDao.java [hunk 11]
```diff
@@ -0,0 +1,38 @@
+package fr.paris.lutece.plugins.example.modules.elastic.extend;
+
+import java.util.ArrayList;
+import java.util.Collection;
+
+import fr.paris.lutece.plugins.elasticdata.business.DataObject;
+import fr.paris.lutece.util.sql.DAOUtil;
+
+public class ExtendProjectDao {
+	private static final String SQL_QUERY_SELECTALL = "SELECT * FROM lutece_test2.extend_extender_hit;";
```
Comment (jonenst) [inline src/java/fr/paris/lutece/plugins/example/modules/elastic/extend/ExtendProjectDao.java]: the database name should not be there

File: src/java/fr/paris/lutece/plugins/example/modules/elastic/extend/ProjectDataObject2.java [hunk 12]
```diff
@@ -0,0 +1,36 @@
...
+     * {@inheritDoc }
+     */
+    @Override
+    public String getTimestamp( )
+    {
+    	String Time="1000";
+		return Time;
+       // time in milliseconds
+    }   
+}
```
Comment (jonenst) [inline src/java/fr/paris/lutece/plugins/example/modules/elastic/extend/ProjectDataObject2.java]: This class should have a better name than ProjectDataObject2

File: src/java/fr/paris/lutece/plugins/example/modules/elastic/extend/ProjectDataSource2.java [hunk 13]
```diff
@@ -0,0 +1,17 @@
...
+
+public class ProjectDataSource2 extends AbstractDataSource implements DataSource{
+	
+	@Override
+    public Collection<DataObject> getDataObjects( )
+    {
+        ExtendProjectDao dao = new ExtendProjectDao();
+        return dao.selectAll();
+    }
+}
```
Comment (jonenst) [inline src/java/fr/paris/lutece/plugins/example/modules/elastic/extend/ProjectDataSource2.java]: This class should have a better name than ProjectDataSource2

File: src/java/fr/paris/lutece/plugins/example/modules/elastic/extend/ProjectDataSource2.java [hunk 14]
```diff
@@ -0,0 +1,17 @@
...
+import fr.paris.lutece.plugins.elasticdata.business.AbstractDataSource;
+import fr.paris.lutece.plugins.elasticdata.business.DataObject;
+import fr.paris.lutece.plugins.elasticdata.business.DataSource;
+
+public class ProjectDataSource2 extends AbstractDataSource implements DataSource{
+	
+	@Override
+    public Collection<DataObject> getDataObjects( )
+    {
+        ExtendProjectDao dao = new ExtendProjectDao();
```
Comment (jonenst) [inline src/java/fr/paris/lutece/plugins/example/modules/elastic/extend/ProjectDataSource2.java]: Same as projectDataSource, inject the dao via spring

File: src/sql/plugins/example/modules/elastic/plugin/create_db_example-elastic.sql [hunk 15]
```diff
@@ -0,0 +1,12 @@
+
+--
+-- Structure for table example_elastic_
+--
+
+DROP TABLE IF EXISTS example_elastic_;
```
Comment (jonenst) [inline src/sql/plugins/example/modules/elastic/plugin/create_db_example-elastic.sql:6]: bad table name

File: webapp/WEB-INF/conf/plugins/example-elastic_context.xml [hunk 16]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<beans xmlns="http://www.springframework.org/schema/beans"
+       xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+       xsi:schemaLocation="http://www.springframework.org/schema/beans http://www.springframework.org/schema/beans/spring-beans-3.0.xsd" >
+
+    <bean id="example-elastic.projectDAO" class="fr.paris.lutece.plugins.example.modules.elastic.business.ProjectDAO" />
+
```
Comment (jonenst) [inline webapp/WEB-INF/conf/plugins/example-elastic_context.xml]: missing the datasources declarations ?

**Comment by jonenst** [comment]
Also, it would be better to make a first commit with the exact output of plugin-wizard. And then a second commit with your changes

File: pom.xml [hunk 17]
```diff
@@ -68,4 +61,39 @@
...
+    									<ignore></ignore>
+    								</action>
+    							</pluginExecution>
+    						</pluginExecutions>
+    					</lifecycleMappingMetadata>
+    				</configuration>
+    			</plugin>
+    		</plugins>
+    	</pluginManagement>
+    </build>
```
Comment (jonenst) [inline pom.xml]: Don't commit eclipse's m2e settings

File: src/java/fr/paris/lutece/plugins/example/modules/elastic/elasticdata/ElasticProjectDao.java [hunk 18]
```diff
@@ -7,7 +7,7 @@
 import fr.paris.lutece.util.sql.DAOUtil;
 
 public class ElasticProjectDao {
-	 private static final String SQL_QUERY_SELECTALL = "SELECT * FROM lutece_test2.example_project;";
+	 private static final String SQL_QUERY_SELECTALL = "SELECT * FROM example_project;";
```
Comment (jonenst) [inline src/java/fr/paris/lutece/plugins/example/modules/elastic/elasticdata/ElasticProjectDao.java]: don't add a semicolon at the end of the string

File: webapp/WEB-INF/conf/plugins/example-elastic_context.xml [hunk 19]
```diff
@@ -5,6 +5,19 @@
...
-
+	
+	<bean id="elasticdata.dataSource" class="fr.paris.lutece.plugins.example.modules.elastic.elasticdata.ProjectDataSource" >
+           <property name="id" value="ProjectDataSource" />
+           <property name="name" value="Project Data Source" />
+           <property name="targetIndexName" value="projectdatasource" />
+           <property name="dataType" value="projectdataobject" />
+       </bean>
+	   
+	   <bean id="elasticdata.dataSource2" class="fr.paris.lutece.plugins.example.modules.elastic.extend.ProjectDataSource2" >
```
Comment (jonenst) [inline webapp/WEB-INF/conf/plugins/example-elastic_context.xml]: bad class name (should be ExtendProjectDataSource ?)

**Comment by Baptistebgh** [comment]
I force push creating a pull-request

**Pull request closed without merging**